### ISSUE TITLE
unit test failure (test_yang_data) on big endian systems (0.16.52)

### ISSUE THREAD

**Issue by eqvinox** [CESNET/libyang#679]
Something seems to be going wrong on `sparc64`, `ppc64` and `hppa` (all big-endian 64-bit systems) specifically in this unit test.  Logfiles:
https://buildd.debian.org/status/fetch.php?pkg=libyang&arch=sparc64&ver=0.16.52-1&stamp=1544025575&raw=0
https://buildd.debian.org/status/fetch.php?pkg=libyang&arch=ppc64&ver=0.16.52-1&stamp=1544026374&raw=0
https://buildd.debian.org/status/fetch.php?pkg=libyang&arch=hppa&ver=0.16.52-1&stamp=1544027739&raw=0

Unfortunately I don't have any further details on this.  I'm adding the `-V` option to the test run to get more info but that will take a few hours.  But considering that it passes tests on `ppc64el` (little-endian mode) it seems extremely likely to be somehow endianness-related.

**Comment by eqvinox** [comment]
Add: also fails on `powerpc` (32bit big-endian)
https://buildd.debian.org/status/fetch.php?pkg=libyang&arch=powerpc&ver=0.16.52-1&stamp=1544031030&raw=0

**Comment by eqvinox** [comment]
on `powerpc`:
```
28: Test command: /home/equinox/libyang/build/tests/test_yang_data
28: Environment variables: 
28:  LIBYANG_EXTENSIONS_PLUGINS_DIR=/home/equinox/libyang/build/src/extensions
28:  LIBYANG_USER_TYPES_PLUGINS_DIR=/home/equinox/libyang/build/src/user_types
28:  MALLOC_CHECK_=3
28: Test timeout computed to be: 9.99988e+06
28: [==========] Running 5 test(s).
28: [ RUN      ] test_new_path
28: libyang[0]: Invalid value "test" in "first" element. (path: /yang-data:#node/test1/first)
28: libyang[0]: Failed to create node "first" as a child of "test1". (path: /yang-data:#node/test1/first)
28: libyang[0]: Schema node not found. (path: /yang-data:test1)
28: [       OK ] test_new_path
28: [ RUN      ] test_new_yangdata
28: [       OK ] test_new_yangdata
28: [ RUN      ] test_path_yangdata
28: [       OK ] test_path_yangdata
28: [ RUN      ] test_validate_yangdata
28: [  ERROR   ] --- Test failed with exception: Segmentation fault(11)
28: [  FAILED  ] test_validate_yangdata
28: [ RUN      ] test_mem_yangdata
28: libyang[1]: String "Bad packet" not freed from the dictionary, refcount 1
28: libyang[1]: String "protocol" not freed from the dictionary, refcount 1
28: [  ERROR   ] --- Test failed with exception: Segmentation fault(11)
28: [  FAILED  ] test_mem_yangdata
28: [==========] 5 test(s) run.
28: [  PASSED  ] 3 test(s).
28: [  FAILED  ] 2 test(s), listed below:
28: [  FAILED  ] test_validate_yangdata
28: [  FAILED  ] test_mem_yangdata
28: 
28:  2 FAILED TEST(S)
 28/108 Test  #28: test_yang_data ...................***Failed    0.10 sec
```

(This is a system I have at home, if you need ssh access please send me an E-Mail with your ssh public key.)

**Comment by eqvinox** [comment]
```
(gdb) run
Starting program: /home/equinox/libyang/build/tests/test_yang_data 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/libthread_db.so.1".
[==========] Running 5 test(s).
[ RUN      ] test_new_path
libyang[0]: Invalid value "test" in "first" element. (path: /yang-data:#node/test1/first)
libyang[0]: Failed to create node "first" as a child of "test1". (path: /yang-data:#node/test1/first)
libyang[0]: Schema node not found. (path: /yang-data:test1)
[       OK ] test_new_path
[ RUN      ] test_new_yangdata
[       OK ] test_new_yangdata
[ RUN      ] test_path_yangdata
[       OK ] test_path_yangdata
[ RUN      ] test_validate_yangdata

Program received signal SIGSEGV, Segmentation fault.
resolve_iffeature (expr=0x25) at /home/equinox/libyang/src/resolve.c:1411
1411	    if (expr->expr) {
(gdb) where
#0  resolve_iffeature (expr=0x25) at /home/equinox/libyang/src/resolve.c:1411
#1  0x003263d0 in lys_is_disabled (node=<optimized out>, recursive=recursive@entry=2) at /home/equinox/libyang/src/tree_schema.c:93
#2  0x002d9220 in lyv_data_context (node=node@entry=0x42ce80, options=options@entry=16777216, unres=unres@entry=0x425220) at /home/equinox/libyang/src/validation.c:63
#3  0x00341e3c in lyd_validate (node=0x4214d8, options=<optimized out>, var_arg=<optimized out>) at /home/equinox/libyang/src/tree_data.c:4965
#4  0x00401338 in test_validate_yangdata (state=<optimized out>) at /home/equinox/libyang/tests/data/test_yang_data.c:194
#5  0x003d57e8 in ?? () from /usr/lib/libcmocka.so.0
#6  0x003d63c0 in _cmocka_run_group_tests () from /usr/lib/libcmocka.so.0
Backtrace stopped: frame did not save the PC
```

**Comment by eqvinox** [comment]
fixed in #680
